src/Drupal/DrupalExtension/Context/BigPipeContext.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Drupal\DrupalExtension\Context;
+
+use Drupal\big_pipe\Render\Placeholder\BigPipeStrategy;
+use Behat\Mink\Exception\UnsupportedDriverActionException;
+
+/**
+ * Big Pipe context.
+ */
+class BigPipeContext extends RawDrupalContext {
+
+    /**
+     * Prepares Big Pipe NOJS cookie if needed.
+     *
+     * @BeforeScenario
+     */
+    public function prepareBigPipeNoJsCookie()
```

Comment:
I did have something like this in a custom project, but it stopped working with behat-extension 4, due to the authentication manager. That now does a fast logout unconditionally before login which kills the session and with that, the cookie.

I actually already did the same in my overriden logout() method with drupal-extension 3 but I thn also re-set the cookie there.

I switched to an @AfterStep and adding the cookie if I have a user and the cookie isn't there yet, but that doesn't seem like a very efficient approach.

Reply:
Maybe we can solve it by emitting an event when we do the fast logout in the authentication manager?